// http-status-kit/dist/index.mjs
// src/status-codes.ts
var statusCodes = /* @__PURE__ */ ((statusCodes2) => {
  statusCodes2[statusCodes2["CONTINUE"] = 100] = "CONTINUE";
  statusCodes2[statusCodes2["SWITCHING_PROTOCOLS"] = 101] = "SWITCHING_PROTOCOLS";
  statusCodes2[statusCodes2["PROCESSING"] = 102] = "PROCESSING";
  statusCodes2[statusCodes2["EARLY_HINTS"] = 103] = "EARLY_HINTS";
  statusCodes2[statusCodes2["OK"] = 200] = "OK";
  statusCodes2[statusCodes2["CREATED"] = 201] = "CREATED";
  statusCodes2[statusCodes2["ACCEPTED"] = 202] = "ACCEPTED";
  statusCodes2[statusCodes2["NON_AUTHORITATIVE_INFORMATION"] = 203] = "NON_AUTHORITATIVE_INFORMATION";
  statusCodes2[statusCodes2["NO_CONTENT"] = 204] = "NO_CONTENT";
  statusCodes2[statusCodes2["RESET_CONTENT"] = 205] = "RESET_CONTENT";
  statusCodes2[statusCodes2["PARTIAL_CONTENT"] = 206] = "PARTIAL_CONTENT";
  statusCodes2[statusCodes2["MULTI_STATUS"] = 207] = "MULTI_STATUS";
  statusCodes2[statusCodes2["ALREADY_REPORTED"] = 208] = "ALREADY_REPORTED";
  statusCodes2[statusCodes2["IM_USED"] = 226] = "IM_USED";
  statusCodes2[statusCodes2["MULTIPLE_CHOICES"] = 300] = "MULTIPLE_CHOICES";
  statusCodes2[statusCodes2["MOVED_PERMANENTLY"] = 301] = "MOVED_PERMANENTLY";
  statusCodes2[statusCodes2["FOUND"] = 302] = "FOUND";
  statusCodes2[statusCodes2["SEE_OTHER"] = 303] = "SEE_OTHER";
  statusCodes2[statusCodes2["NOT_MODIFIED"] = 304] = "NOT_MODIFIED";
  statusCodes2[statusCodes2["USE_PROXY"] = 305] = "USE_PROXY";
  statusCodes2[statusCodes2["TEMPORARY_REDIRECT"] = 307] = "TEMPORARY_REDIRECT";
  statusCodes2[statusCodes2["PERMANENT_REDIRECT"] = 308] = "PERMANENT_REDIRECT";
  statusCodes2[statusCodes2["BAD_REQUEST"] = 400] = "BAD_REQUEST";
  statusCodes2[statusCodes2["UNAUTHORIZED"] = 401] = "UNAUTHORIZED";
  statusCodes2[statusCodes2["PAYMENT_REQUIRED"] = 402] = "PAYMENT_REQUIRED";
  statusCodes2[statusCodes2["FORBIDDEN"] = 403] = "FORBIDDEN";
  statusCodes2[statusCodes2["NOT_FOUND"] = 404] = "NOT_FOUND";
  statusCodes2[statusCodes2["METHOD_NOT_ALLOWED"] = 405] = "METHOD_NOT_ALLOWED";
  statusCodes2[statusCodes2["NOT_ACCEPTABLE"] = 406] = "NOT_ACCEPTABLE";
  statusCodes2[statusCodes2["PROXY_AUTHENTICATION_REQUIRED"] = 407] = "PROXY_AUTHENTICATION_REQUIRED";
  statusCodes2[statusCodes2["REQUEST_TIMEOUT"] = 408] = "REQUEST_TIMEOUT";
  statusCodes2[statusCodes2["CONFLICT"] = 409] = "CONFLICT";
  statusCodes2[statusCodes2["GONE"] = 410] = "GONE";
  statusCodes2[statusCodes2["LENGTH_REQUIRED"] = 411] = "LENGTH_REQUIRED";
  statusCodes2[statusCodes2["PRECONDITION_FAILED"] = 412] = "PRECONDITION_FAILED";
  statusCodes2[statusCodes2["PAYLOAD_TOO_LARGE"] = 413] = "PAYLOAD_TOO_LARGE";
  statusCodes2[statusCodes2["URI_TOO_LONG"] = 414] = "URI_TOO_LONG";
  statusCodes2[statusCodes2["UNSUPPORTED_MEDIA_TYPE"] = 415] = "UNSUPPORTED_MEDIA_TYPE";
  statusCodes2[statusCodes2["RANGE_NOT_SATISFIABLE"] = 416] = "RANGE_NOT_SATISFIABLE";
  statusCodes2[statusCodes2["EXPECTATION_FAILED"] = 417] = "EXPECTATION_FAILED";
  statusCodes2[statusCodes2["IM_A_TEAPOT"] = 418] = "IM_A_TEAPOT";
  statusCodes2[statusCodes2["MISDIRECTED_REQUEST"] = 421] = "MISDIRECTED_REQUEST";
  statusCodes2[statusCodes2["UNPROCESSABLE_ENTITY"] = 422] = "UNPROCESSABLE_ENTITY";
  statusCodes2[statusCodes2["LOCKED"] = 423] = "LOCKED";
  statusCodes2[statusCodes2["FAILED_DEPENDENCY"] = 424] = "FAILED_DEPENDENCY";
  statusCodes2[statusCodes2["TOO_EARLY"] = 425] = "TOO_EARLY";
  statusCodes2[statusCodes2["UPGRADE_REQUIRED"] = 426] = "UPGRADE_REQUIRED";
  statusCodes2[statusCodes2["PRECONDITION_REQUIRED"] = 428] = "PRECONDITION_REQUIRED";
  statusCodes2[statusCodes2["TOO_MANY_REQUESTS"] = 429] = "TOO_MANY_REQUESTS";
  statusCodes2[statusCodes2["REQUEST_HEADER_FIELDS_TOO_LARGE"] = 431] = "REQUEST_HEADER_FIELDS_TOO_LARGE";
  statusCodes2[statusCodes2["UNAVAILABLE_FOR_LEGAL_REASONS"] = 451] = "UNAVAILABLE_FOR_LEGAL_REASONS";
  statusCodes2[statusCodes2["INTERNAL_SERVER_ERROR"] = 500] = "INTERNAL_SERVER_ERROR";
  statusCodes2[statusCodes2["NOT_IMPLEMENTED"] = 501] = "NOT_IMPLEMENTED";
  statusCodes2[statusCodes2["BAD_GATEWAY"] = 502] = "BAD_GATEWAY";
  statusCodes2[statusCodes2["SERVICE_UNAVAILABLE"] = 503] = "SERVICE_UNAVAILABLE";
  statusCodes2[statusCodes2["GATEWAY_TIMEOUT"] = 504] = "GATEWAY_TIMEOUT";
  statusCodes2[statusCodes2["HTTP_VERSION_NOT_SUPPORTED"] = 505] = "HTTP_VERSION_NOT_SUPPORTED";
  statusCodes2[statusCodes2["VARIANT_ALSO_NEGOTIATES"] = 506] = "VARIANT_ALSO_NEGOTIATES";
  statusCodes2[statusCodes2["INSUFFICIENT_STORAGE"] = 507] = "INSUFFICIENT_STORAGE";
  statusCodes2[statusCodes2["LOOP_DETECTED"] = 508] = "LOOP_DETECTED";
  statusCodes2[statusCodes2["NOT_EXTENDED"] = 510] = "NOT_EXTENDED";
  return statusCodes2;
})(statusCodes || {});

// src/status-messages.ts
var statusMessages = /* @__PURE__ */ ((statusMessages2) => {
  statusMessages2["CONTINUE"] = "Continue sending data.";
  statusMessages2["SWITCHING_PROTOCOLS"] = "Switching protocols as requested.";
  statusMessages2["PROCESSING"] = "The server has received the request and is still processing it.";
  statusMessages2["EARLY_HINTS"] = "The server is returning some headers before the final response.";
  statusMessages2["OK"] = "Success! Your request has been completed.";
  statusMessages2["CREATED"] = "Success! New resource created.";
  statusMessages2["ACCEPTED"] = "Request received! We're working on it.";
  statusMessages2["NON_AUTHORITATIVE_INFORMATION"] = "The request was successful but the information may be from a third party.";
  statusMessages2["NO_CONTENT"] = "Success! No additional content available.";
  statusMessages2["RESET_CONTENT"] = "The request was successful, please reset the document view.";
  statusMessages2["PARTIAL_CONTENT"] = "The server is delivering only part of the resource due to a range header sent by the client.";
  statusMessages2["MULTI_STATUS"] = "The message body contains multiple status codes for separate operations (WebDAV).";
  statusMessages2["ALREADY_REPORTED"] = "The members of a DAV binding have already been enumerated (WebDAV).";
  statusMessages2["IM_USED"] = "The server has fulfilled the request for the resource, and the response is a representation of the result of one or more instance manipulations applied to the current instance.";
  statusMessages2["MULTIPLE_CHOICES"] = "Multiple options available. Please choose one.";
  statusMessages2["MOVED_PERMANENTLY"] = "The requested resource has been permanently moved to a new URL.";
  statusMessages2["FOUND"] = "Resource temporarily moved. Please wait while we redirect you...";
  statusMessages2["SEE_OTHER"] = "The answer is accessible via a different URI with a GET request.";
  statusMessages2["NOT_MODIFIED"] = "The resource is unchanged since the last request.";
  statusMessages2["USE_PROXY"] = "The requested resource must be accessed through the proxy.";
  statusMessages2["TEMPORARY_REDIRECT"] = "The request should be repeated with another URI temporarily.";
  statusMessages2["PERMANENT_REDIRECT"] = "The request and future requests should be repeated with another URI permanently.";
  statusMessages2["BAD_REQUEST"] = "Request not understood due to syntax error.";
  statusMessages2["UNAUTHORIZED"] = "You must authenticate to proceed.";
  statusMessages2["PAYMENT_REQUIRED"] = "Payment is required to access the requested resource.";
  statusMessages2["FORBIDDEN"] = "You don't have permission to view this.";
  statusMessages2["NOT_FOUND"] = "Requested resource unavailable.";
  statusMessages2["METHOD_NOT_ALLOWED"] = "The method specified in the request is not allowed for the resource.";
  statusMessages2["NOT_ACCEPTABLE"] = "Requested resource can't provide acceptable content.";
  statusMessages2["PROXY_AUTHENTICATION_REQUIRED"] = "The client must authenticate itself with the proxy.";
  statusMessages2["REQUEST_TIMEOUT"] = "The server timed out waiting for the request.";
  statusMessages2["CONFLICT"] = "The request conflicts with the current state of the resource.";
  statusMessages2["GONE"] = "The requested resource is no longer available and will not be available again.";
  statusMessages2["LENGTH_REQUIRED"] = "The server requires the Content-Length header to be sent with the request.";
  statusMessages2["PRECONDITION_FAILED"] = "One or more conditions given in the request header fields were evaluated as false.";
  statusMessages2["PAYLOAD_TOO_LARGE"] = "The request is larger than the server is willing or able to process.";
  statusMessages2["URI_TOO_LONG"] = "The URI provided was too long for the server to process.";
  statusMessages2["UNSUPPORTED_MEDIA_TYPE"] = "The media format of the requested data is not supported by the server.";
  statusMessages2["RANGE_NOT_SATISFIABLE"] = "The range specified by the Range header field in the request can't be fulfilled.";
  statusMessages2["EXPECTATION_FAILED"] = "The expectation given in the request header could not be met by the server.";
  statusMessages2["IM_A_TEAPOT"] = "The server refuses the attempt to brew coffee with a teapot.";
  statusMessages2["MISDIRECTED_REQUEST"] = "The request was directed at a server that is not able to produce a response.";
  statusMessages2["UNPROCESSABLE_ENTITY"] = "The request was well-formed but was unable to be followed due to semantic errors.";
  statusMessages2["LOCKED"] = "The resource that is being accessed is locked.";
  statusMessages2["FAILED_DEPENDENCY"] = "The request failed due to failure of a previous request.";
  statusMessages2["TOO_EARLY"] = "The server is unwilling to risk processing a request that might be replayed.";
  statusMessages2["UPGRADE_REQUIRED"] = "The client should switch to a different protocol.";
  statusMessages2["PRECONDITION_REQUIRED"] = "The origin server requires the request to be conditional.";
  statusMessages2["TOO_MANY_REQUESTS"] = "You've sent too many requests.";
  statusMessages2["REQUEST_HEADER_FIELDS_TOO_LARGE"] = "The server is unwilling to process the request because its header fields are too large.";
  statusMessages2["UNAVAILABLE_FOR_LEGAL_REASONS"] = "The server is denying access to the resource as a consequence of a legal demand.";
  statusMessages2["INTERNAL_SERVER_ERROR"] = "An unexpected error occurred.";
  statusMessages2["NOT_IMPLEMENTED"] = "The server does not support the functionality required to fulfill the request.";
  statusMessages2["BAD_GATEWAY"] = "The server, while acting as a gateway or proxy, received an invalid response from the upstream server.";
  statusMessages2["SERVICE_UNAVAILABLE"] = "Server not ready, try again later.";
  statusMessages2["GATEWAY_TIMEOUT"] = "The server, while acting as a gateway or proxy, did not receive a timely response from the upstream server.";
  statusMessages2["HTTP_VERSION_NOT_SUPPORTED"] = "The server does not support the HTTP protocol version that was used in the request.";
  statusMessages2["VARIANT_ALSO_NEGOTIATES"] = "The server has an internal configuration error and needs to negotiate a different variant.";
  statusMessages2["INSUFFICIENT_STORAGE"] = "The server is unable to store the representation needed to complete the request.";
  statusMessages2["LOOP_DETECTED"] = "The server detected an infinite loop while processing the request (WebDAV).";
  statusMessages2["NOT_EXTENDED"] = "Further extensions to the request are required for the server to fulfill it.";
  return statusMessages2;
})(statusMessages || {});
export {
  statusCodes,
  statusMessages
};
